internal/impl/packages.go

```diff
@@ -444,22 +443,31 @@ func resetProfileDirForFlakes(profileDir string) (err error) {
 	}()
 
 	dir, err := filepath.EvalSymlinks(profileDir)
+	if errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
 	if err != nil {
 		return errors.WithStack(err)
 	}
 
-	needsReset := false
 	if featureflag.Flakes.Enabled() {
 		// older nix profiles have a manifest.nix file present
-		needsReset = fileutil.Exists(filepath.Join(dir, "manifest.nix"))
+		_, err := os.Stat(filepath.Join(dir, "manifest.nix"))
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		if err != nil {
+			return errors.WithStack(err)
+		}
 	} else {
 		// newer flake nix profiles have a manifest.json file present
-		needsReset = fileutil.Exists(filepath.Join(dir, "manifest.json"))
-	}
-
-	if !needsReset {
-		return nil
+		_, err := os.Stat(filepath.Join(dir, "manifest.json"))
+		if errors.Is(err, os.ErrNotExist) {
+			return nil
+		}
+		if err != nil {
+			return errors.WithStack(err)
+		}
```

Comment:
I understand the changes in lines 446-448

Could we push this fix down inside `fileutil.Exists`? Using that api is easier to read (albeit masking a genuine error from `os.Stat`)

Reply:
Do we want to mask the error? If there's something like a permissions error, then I figured returning that error would be helpful.

Reply:
that's a fair point....